drivers/SmartThings/sonos/src/init.lua

```diff
@@ -304,9 +380,31 @@ local driver = Driver("Sonos", {
     local dni_normalized = dni:gsub("-", ""):gsub(":", ""):lower()
     local other_normalized = other:gsub("-", ""):gsub(":", ""):lower()
     return dni_normalized == other_normalized
+  end,
+  get_device_by_dni = function(self, dni)
+    local device_uuid = self.datastore.dni_to_device_id[dni]
+    if not device_uuid then return nil end
+    return self:get_device_info(device_uuid)
   end
 })
 
+if driver.datastore["_field_cache"] == nil then
+  driver.datastore["_field_cache"] = {}
+end
+
+if driver.datastore["dni_to_device_id"] == nil then
+  driver.datastore["dni_to_device_id"] = {}
+end
+
+driver._field_cache = driver.datastore._field_cache
+
+-- Kick off a scan right away to attempt to populate some information
+driver:call_with_delay(3, scan_for_ssdp_updates, "Sonos SSDP Initial Scan")
+
+-- re-scan every minute
+local SSDP_SCAN_INTERVAL_SECONDS = 60
```

Comment:
Any reason for this value? It seems aggressive to me. I know for most network setups, IPs are not changing that much if at all, so I would think it would be more like once or twice a day rather than every minute.

Reply:
I had actually intended to set this to 10 minutes, I was using 1 minute in testing.